### PR TITLE
perf(lsp): format in a blocking task

### PR DESCRIPTION
**Pull request by dsherret**
Just noticed this while I was looking into a different lsp formatting issue. We actually used to have this, but I mistakenly removed it during a large refactor.

### CODE REVIEW

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -1638,24 +1638,33 @@ impl Inner {
...
+              .unwrap_or(file_path);
+            // it's not a js/ts file, so attempt to format its contents
+            format_file(&file_path, &document.content(), &fmt_options)
+          }
+        }
       }
-    };
+    })
+    .await
+    .unwrap();
```
Comment (bartlomieju) [inline cli/lsp/language_server.rs:1667]: This unwrap can cause panics, shouldn't it be `?`?

Comment (dsherret) [reply cli/lsp/language_server.rs:1667]: Isn't that only if the task panics?

Comment (bartlomieju) [reply cli/lsp/language_server.rs:1667]: Oh sorry 🤦 you're handling the result later

**Review by bartlomieju — APPROVED** [review]

File: cli/lsp/language_server.rs (hunk 1 above)
Comment (dsherret) [reply cli/lsp/language_server.rs:1667]: No problem!

**Pull request merged**